rest/rule_test.go

```diff
@@ -987,6 +987,55 @@ func TestApplyInboundRulesNestedRulesSlice(t *testing.T) {
 	assert.Nil(err, "Error should be nil")
 }
 
+// Ensures that nested inbound Rules marking primitives fail if type cannot be coerced.
+func TestApplyInboundRulesNestedRulesUnableToCoerceToPrimitive(t *testing.T) {
+	assert := assert.New(t)
+	payload := Payload{"foo": []interface{}{"bar", []interface{}{"baz"}}}
+	rules := NewRules((*TestResourceSlice)(nil),
+		&Rule{
+			Field: "Foo",
+			FieldAlias: "foo",
+			Type: Slice,
+			Versions: []string{"1"},
+			Rules: NewRules((*string)(nil),
+				&Rule{
+					Type: String,
+				},
+			),
+		},
+	)
+	expectedErr := fmt.Errorf("Unable to coerce slice to map[string]interface{}")
```

Comment:
Can you explain why this is the expected error? Isn't it trying to coerce a slice to a string?